events/importer/espoo.py

```
@@ -1,800 +1,534 @@
+    }
+
+    for instance in instances:
+        post_data = {}
+        for field_name, mapper in post_field_mappers.items():
+            obj_data = instance_data_map[instance]
+            post_data = mapper(
+                field_name, obj_data[field_name], post_data, instance, instance_mapping
+            )
+
+        if post_data:
+            type(instance).objects.filter(pk=instance.pk).update(**post_data)
+
+    return instance_mapping, syncher
+
+
+def _split_common_objs(
+    model: Type[Model],
+    data_source: DataSource,
+    origin_objs: list[dict],
+) -> tuple[list[Model], list[dict]]:
+    """
+    Generic method for discovering origin objects that can be matched
+    to existing models in other data sources. For example tprek entries
+    should generally be identical between Helsinki LE and Espoo LE.
+    """
+    origin_obj_ids = set(obj["id"] for obj in origin_objs)
+    common_objs = list(
+        model.objects.filter(id__in=origin_obj_ids)
+        .exclude(data_source=data_source)
+        .only("id")
+    )
+    common_obj_ids = [org.id for org in common_objs]
+    origin_objs = [obj for obj in origin_objs if obj["id"] not in common_obj_ids]
+    return common_objs, origin_objs
+
+
+def _add_id_to_set(container: set, obj, key, jsonld=False):
+    """
+    Utility function for adding ids to a set
+    """
+    if value := obj.get(key):
+        if not isinstance(value, list):
+            value = [value]
+        if jsonld:
+            value = parse_jsonld_ids(value)
+        container.update(value)
+
+
+def _add_id_to_dict(container: dict, obj, key, jsonld=False):
+    """
+    Utility function for adding ids to a dict
+    """
+    if value := obj.get(key):
+        if not isinstance(value, list):
+            value = [value]
+        for row in value:
+            if jsonld:
+                row_id = parse_jsonld_id(row)
+            else:
+                row_id = row["id"]
+            container[row_id] = row
```

As discussed, unless there is a good reason why these need to be like this, it would be better if these didn't silently ignore missing keys.